package.json

```diff
@@ -40,7 +40,8 @@
     "fs-plus": "^3.0.0"
   },
   "scripts": {
-    "prepublish": "standard src spec",
-    "test": "jasmine-focused --captureExceptions spec"
+    "lint": "standard src spec",
+    "test": "jasmine-focused --captureExceptions spec",
+    "prepare": "git submodule update --init --recursive"
```

Comment:
I tried this on my machine, and `prepare` (surprisingly) doesn't run before `npm install`, so building errors out with the missing submodule. So `prepare` is not better than `prepublishOnly` in this case.

`prepare` is officially [deprecated](), so perhaps this can use `prepublishOnly` instead of `prepare`.

Reply:
If we want this to run before the package is installed, we could do that in `preinstall`...

<details><summary>Proposed way to do this in a preinstall script (click to expand)</summary>

The `preinstall` scripts runs even for folks who download this package from `npmjs.com`, where we know the submodule will be initialized. It would be nice to check if the submodule files are present (maybe check if `deps/libgit2/README.md` exists), and only initialize if it's not present. This check would have to be in a NodeJS script so it can run on all platforms: Linux, macOS and Windows. 

So if it's worth doing all that, I could try to code that up.

</details>

Reply:
Prepare is not deprecated. It is the recommended way to do something related to the building.
https://docs.npmjs.com/cli/v6/using-npm/scripts#life-cycle-scripts

Reply:
Sorry I forgot to paste the URL. This section says it is deprecated: https://docs.npmjs.com/cli/v6/using-npm/scripts#prepare-and-prepublish

---

> Prepare works before `npm publish` and on local top-level `npm install` (during development), but prepublishOnly only works in `npm publish`.

True. But I was surprised the `prepublish` script runs pretty late. If you don't have the submodule initialized before you do `npm install` locally, it will try to build the native code, and error out.

<details><summary>Error message (click to expand)</summary>

```console
% git status
On branch submodule-init
[...]
% npm install  

> git-utils@5.7.0 install /Users/[user]/git-utils
> node-gyp rebuild

make: *** No rule to make target `Release/obj.target/libgit2/deps/libgit2/src/annotated_commit.o', needed by `Release/git2.a'.  Stop.
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/[user]/n-prefix/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/Users/[user]/n-prefix/bin/node" "/Users/[user]/n-prefix/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/[user]/git-utils
gyp ERR! node -v v12.20.0
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! git-utils@5.7.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the git-utils@5.7.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/[user]/.npm/_logs/2020-12-30T23_59_57_507Z-debug.log
```

</details>

---

So as I was saying, there is really no way to get this to _usefully_ run as a script for local installs... other than having a script to check the content of `deps/libgit2` and see if it's initialized, and if it's not initialized, then initialize it.

Something like this:

```js
if ! fs.existsSync('deps/libgit2/README.md) {
  spawnSync('git submodule update --init');
};
```

Reply:
So there's two options:
1. Give up on automatically initializing the submodule for local `npm install` in this repo
   i. This is totally reasonable. This is how it was handled at the other repos.
1. Write a JS script that initializes it for you, and run that in `preinstall`.

Reply:
Oops, you are right... `prepare` isn't deprecated.

It still doesn't actually run early enough to prevent the `node-gyp` build error.

Reply:
https://github.com/atom/git-utils/pull/104/commits/409bff86d02bb685d403913294c7f13946f8fe1e

Reply:
Well, that works. 👍